File: 0283-move-zeroes/0283-move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int l=0;
    int r=0;
    int t;
    while (l < numsSize){
        if (*(nums+l)==0)
            break;
        l++;
    }
    r=l+1;
    while ( r < numsSize){
        if (*(nums+r)!=0){
            t=*(nums+r);
            *(nums+r)=*(nums+l);
            *(nums+l)=t;
            l++;


        }
        r++;


    }
    
}

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int left=0;
        int right=0;
        int n=nums.length;
        while(left < n && nums[left]!=0){
            left++;
            
        }
        right=left+1;
        while(right< n){
            if(nums[right]==0){
                right++;
            }
            else{
                int t=nums[left];
                nums[left]=nums[right];
                nums[right]=t;
                left++;
                right++;
            }

        }
        
    }
}